### PR TITLE
Feature/LGA-530 welsh translations

### PR DESCRIPTION
**Pull request by joshlow**
## What does this pull request do?

This PR helps make the service fully translated into Welsh by adding missing Trans tags and the corresponding Welsh translations.

## Any other changes that would benefit highlighting?

There have also been fixes to the tag format so no markup is partially included outside/inside translation tags.

This improves coverage of the Welsh translations. During the process we have discovered other missing translations.

## Checklist

This is ticket LGA-530
https://dsdmoj.atlassian.net/browse/LGA-530

### CODE REVIEW

File: cla_public/static-templates/errors/50x.html [hunk 1]
```diff
@@ -80,14 +80,14 @@
...
-          <a href="#main-content" class="govuk-skip-link">Skip to main content</a>
+          <a href="#main-content" class="govuk-skip-link">{% trans %}Skip to main content{% endtrans %}</a>
         </div>
       </div>
 
       <div id="global-cookie-message">
         <div class="outer-block">
           <div class="inner-block">
-            <p>GOV.UK uses cookies to make the site simpler. <a href="https://www.gov.uk/help/cookies">Find out more about cookies</a></p>
+            <p>{% trans %}GOV.UK uses cookies to make the site simpler. <a href="https://www.gov.uk/help/cookies">Find out more about cookies{% endtrans %}</a></p>
```
Comment (MalcolmVonMoJ) [inline cla_public/static-templates/errors/50x.html]: Can we perhaps split this up?
`{% trans %}GOV.UK uses cookies to make the site simpler.{% endtrand %} <a href="https://www.gov.uk/help/cookies">{% trans %}Find out more about cookies{% endtrans %}</a>`
That way we don't need the tag in the translation file.

File: cla_public/static-templates/errors/50x.html [hunk 2]
```diff
@@ -121,51 +121,51 @@
         <div class="govuk-phase-banner">
           <p class="govuk-phase-banner__content">
             <span class="govuk-phase-banner__text" id="aria-feedback">
-              <a href="{{ url_for('base.reasons_for_contacting')}}">Report a problem</a> if you need help or would like to give feedback to improve this service
+              <a href="{{ url_for('base.reasons_for_contacting')}}">{% trans %}Report a problem</a> if you need help or would like to give feedback to improve this service{% endtrans %}
```
Comment (MalcolmVonMoJ) [inline cla_public/static-templates/errors/50x.html]: Are we sure that, in Welsh, the linked part of the text will always be at the front?  If not - we should include the tag in the translation file.  
`{% trans %}<a href="{{ url_for('base.reasons_for_contacting')}}">Report a problem</a> if you need help or would like to give feedback to improve this service{% endtrans %}
That way, the Welsh could be xxxxxxxxx <a>xxxxxxx</a> xxxxxxx if need be.
            `

File: cla_public/static-templates/errors/50x.html [hunk 3]
```diff
@@ -121,51 +121,51 @@
...
-          <h1 class="govuk-heading-xl">This service is temporarily unavailable</h1>
-          <p class="govuk-body">This service is not available right now, but we're working hard to get things up and running again.</p>
-          <p class="govuk-body">You can try again later or contact us. </p>
-          <p class="govuk-body">Telephone:<br />
-            <strong class="govuk-!-font-weight-bold">0345 345 4345</strong>
+          <h1 class="govuk-heading-xl">{% trans %}This service is temporarily unavailable{% endtrans %}</h1>
+          <p class="govuk-body">{% trans %}This service is not available right now, but we're working hard to get things up and running again.{% endtrans %}</p>
+          <p class="govuk-body">{% trans %}You can try again later or contact us.{% endtrans %}</p>
+          <p class="govuk-body">{% trans %}Telephone:<br/>
+            <strong class="govuk-!-font-weight-bold">0345 345 4345{% endtrans %}</strong>
```
Comment (MalcolmVonMoJ) [inline cla_public/static-templates/errors/50x.html]: Again, I'd omit the HTML and telephone number from the translation - the only translatable element here is telephone.  
`<p class="govuk-body">{% trans %}Telephone{% endtrans %}:<br/>
            <strong class="govuk-!-font-weight-bold">0345 345 4345</strong>`

File: cla_public/static-templates/errors/50x.html [hunk 4]
```diff
@@ -121,51 +121,51 @@
...
           </p>
-          <p class="govuk-body">Minicom:<br />
-            <strong class="govuk-!-font-weight-bold">0345 609 6677</strong>
+          <p class="govuk-body">{% trans %}Minicom:<br/>
+            <strong class="govuk-!-font-weight-bold">0345 609 6677{% endtrans %}</strong>
```
Comment (MalcolmVonMoJ) [inline cla_public/static-templates/errors/50x.html]: As above, only translate Minicom

File: cla_public/static-templates/errors/50x.html [hunk 5]
```diff
@@ -121,51 +121,51 @@
...
           </p>
-          <p class="govuk-body">Textphone:<br />
-            <strong class="govuk-!-font-weight-bold">0845 609 6677</strong>
+          <p class="govuk-body">{% trans %}Textphone:<br/>
+            <strong class="govuk-!-font-weight-bold">0845 609 6677{% endtrans %}</strong>
```
Comment (MalcolmVonMoJ) [inline cla_public/static-templates/errors/50x.html]: As above

File: cla_public/static-templates/errors/50x.html [hunk 6]
```diff
@@ -121,51 +121,51 @@
...
           </p>
           <p class="govuk-body">
-            <a href="https://www.gov.uk/call-charges">Find out about call charges here</a>.
+            <a href="https://www.gov.uk/call-charges">{% trans %}Find out about call charges here{% endtrans %}</a>.
           </p>
-          <p class="govuk-body">Opening times:<br />
-            <strong class="govuk-!-font-weight-bold">Monday to Friday: 9am to 8pm<br />Saturday: 9am to 12.30pm</strong>
+          <p class="govuk-body">{% trans %}Opening times:<br />
+            <strong class="govuk-!-font-weight-bold">Monday to Friday: 9am to 8pm<br />Saturday: 9am to 12.30pm{% endtrans %}</strong>
```
Comment (MalcolmVonMoJ) [inline cla_public/static-templates/errors/50x.html]: I would split it up to remove the HTML from the translation:
`<p class="govuk-body">{% trans %}Opening times{% endtrans %}:<br />`
and
`<strong class="govuk-!-font-weight-bold">{% trans %}Monday to Friday: 9am to 8pm<br />Saturday: 9am to 12.30pm{% endtrans %}</strong>`

File: cla_public/static-templates/errors/50x.html [hunk 7]
```diff
@@ -121,51 +122,51 @@
...
-                <a class="govuk-footer__link"  href="/cookies">Cookies</a>
+                <a class="govuk-footer__link"  href="/cookies">{% trans %}Cookies{% endtrans %}</a>
               </li>
               <li class="govuk-footer__inline-list-item">
-                <a class="govuk-footer__link"  href="/reasons-for-contacting">Contact us</a>
+                <a class="govuk-footer__link"  href="/reasons-for-contacting">{% trans %}Contact us{% endtrans %}</a>
               </li>
               <li class="govuk-footer__inline-list-item">
-                <a class="govuk-footer__link"  class="govuk-footer__link" href="https://mojdigital.blog.gov.uk/"><abbr title="Ministry of Justice">MoJ</abbr> Digital Services</a>
+                <a class="govuk-footer__link"  class="govuk-footer__link" href="https://mojdigital.blog.gov.uk/"><abbr title="Ministry of Justice">{% trans %}MoJ</abbr> Digital Services% endtrans %}</a>
```
Comment (MalcolmVonMoJ) [inline cla_public/static-templates/errors/50x.html]: Is that a missing { here?

I think the Welsh translation should include the abbr tag - so we can translate the "Ministry of Justice" bit as well.

Or, maybe separated:
`<a class="govuk-footer__link"  class="govuk-footer__link" href="https://mojdigital.blog.gov.uk/"><abbr title="{% trans %}Ministry of Justice{% endtrans %}">{% trans %}MoJ</abbr> Digital Services{% endtrans %}</a>`

File: cla_public/templates/privacy.html [hunk 8]
```diff
@@ -622,7 +622,7 @@ <h2>{{ _('Complaints') }}</h2>
         <span class="locality">Wilmslow</span>
         <span class="locality">Cheshire</span>
```
Comment (MalcolmVonMoJ) [inline cla_public/templates/privacy.html]: Just an observation: English counties almost always have different Welsh names.  In this case **Swydd Gaer**.  Do we want to bother with that?

Comment (joshlow) [reply cla_public/templates/privacy.html]: @MalcolmVonMoJ - I ran this text through translation and the translators opted to keep it in English, maybe the Post Office would have trouble sending post??

File: cla_public/static-templates/errors/50x.html (hunk 7 above)
Comment (joshlow) [reply cla_public/static-templates/errors/50x.html]: @MalcolmVonMoJ - thanks, good spot. Will move the abbr tag into the translation.

File: cla_public/templates/checker/result/face-to-face.html [hunk 9]
```diff
@@ -67,9 +67,8 @@ <h2 class="govuk-heading-m">{{ _('What happens next') }}</h2>
     </p>
 
     <p class="govuk-body">
-        <a class="govuk-link" href="https://www.gov.uk/done/check-if-civil-legal-advice-can-help-you" aria-labelledby="aria-satisfaction-survey">
-          {% trans %}What did you think of this service?{% endtrans %}
-        </a> ({% trans %}takes 30 seconds{% endtrans %})
+        {% trans %}<a class="govuk-link" href="https://www.gov.uk/done/check-if-civil-legal-advice-can-help-you" aria-labelledby="aria-satisfaction-survey">
+          What did you think of this service?</a> (takes 30 seconds){% endtrans %}
```
Comment (said-moj) [inline cla_public/templates/checker/result/face-to-face.html]: Can this be
`<a class="govuk-link" href="https://www.gov.uk/done/check-if-civil-legal-advice-can-help-you" aria-labelledby="aria-satisfaction-survey">{% trans %}What did you think of this service?{% endtrans %}</a> {% trans %}(takes 30 seconds){% endtrans %}`

File: cla_public/templates/checker/result/ineligible.html [hunk 10]
```diff
@@ -64,7 +64,7 @@ <h1 class="page-title">{{ _('You’re unlikely to get legal aid') }}</h1>
 
        <p>
       {% trans %}
-          <a href="https://www.gov.uk/done/check-if-civil-legal-advice-can-help-you" aria-labelledby="aria-satisfaction-survey">What did you think of this service?</a> (takes 30 seconds)
+          {% trans %}<a href="https://www.gov.uk/done/check-if-civil-legal-advice-can-help-you" aria-labelledby="aria-satisfaction-survey">What did you think of this service?</a> (takes 30 seconds){% endtrans %}
```
Comment (said-moj) [inline cla_public/templates/checker/result/ineligible.html]: Same as above

File: cla_public/templates/scope/mediation.html [hunk 11]
```diff
@@ -17,7 +17,7 @@ <h1 class="page-title">{{ _('You may be able to get legal aid for family mediati
 
        <p>
       {% trans %}
-          <a href="https://www.gov.uk/done/check-if-civil-legal-advice-can-help-you" aria-labelledby="aria-satisfaction-survey">What did you think of this service?</a> (takes 30 seconds)
+          {% trans %}<a href="https://www.gov.uk/done/check-if-civil-legal-advice-can-help-you" aria-labelledby="aria-satisfaction-survey">What did you think of this service?</a> (takes 30 seconds){% endtrans %}
```
Comment (said-moj) [inline cla_public/templates/scope/mediation.html]: Same as above

File: cla_public/static-templates/errors/50x.html [hunk 12]
```diff
@@ -121,51 +122,51 @@
...
           </p>
-          <p class="govuk-body">Opening times:<br />
-            <strong class="govuk-!-font-weight-bold">Monday to Friday: 9am to 8pm<br />Saturday: 9am to 12.30pm</strong>
+          <p class="govuk-body">{% trans %}Opening times:{% endtrans %}<br />
+            <strong class="govuk-!-font-weight-bold">{% trans %}Monday to Friday: 9am to 8pm{% endtrans %}<br />{% trans %}Saturday: 9am to 12.30pm{% endtrans %}</strong>
           </p>
-          <div class="govuk-inset-text">
-            This service is only available if you are a resident of England or Wales.
+          <div class="govuk-inset-text">{% trans %}
+            This service is only available if you are a resident of England or Wales.{% endtrans %}
```
Comment (MalcolmVonMoJ) [inline cla_public/static-templates/errors/50x.html]: Can we have this on one line?
`{% trans %}This service is only available if you are a resident of England or Wales.{% endtrans %}`

File: cla_public/templates/_cookie-consent-form.html [hunk 13]
```diff
@@ -1,97 +1,97 @@
...
     <div class="gem-c-govspeak govuk-govspeak " data-module="govspeak">
-      <h2 class="govuk-heading-l">Strictly necessary cookies</h2>
-      <p class="govuk-body">We use essential cookies to help you use the ‘Check you can get legal aid’ service. These do things such as:</p>
-      <ul class="govuk-list">
-        <li>remember your progress through our service</li>
-        <li>remember you have seen the cookies message</li>
-        <li>remember the language you have selected to view our service</li>
+      <h2 class="govuk-heading-l">{% trans %}Strictly necessary cookies{% endtrans %}</h2>
+      <p class="govuk-body">{% trans %}We use essential cookies to help you use the ‘Check you can get legal aid’ service. These do things such as:{% endtrans %}</p>
+      <ul>
```
Comment (MalcolmVonMoJ) [inline cla_public/templates/_cookie-consent-form.html]: This class shouldn't have been removed.

File: cla_public/templates/_cookie-consent-form.html [hunk 14]
```diff
@@ -1,97 +1,97 @@
...
 </div>
 
 <div class="cookie-settings__form-wrapper">
-	<p class="govuk-body">We use different types of cookies. You can choose which cookies you're happy for us to use and how the data is shared.</p>
+	<p class="govuk-body">{% trans %}We use different types of cookies. You can choose which cookies you're happy for us to use and how the data is shared.{% endtrans %}</p>
 	<form method="get" data-module="cookie-settings">
     <div class="gem-c-govspeak govuk-govspeak " data-module="govspeak">
-      <h2 class="govuk-heading-l">Strictly necessary cookies</h2>
-      <p class="govuk-body">We use essential cookies to help you use the ‘Check you can get legal aid’ service. These do things such as:</p>
-      <ul class="govuk-list">
```
Comment (MalcolmVonMoJ) [inline cla_public/templates/_cookie-consent-form.html]: This class shouldn't have been removed.

File: cla_public/templates/_cookie-consent-form.html [hunk 15]
```diff
@@ -1,97 +1,97 @@
...
     <div class="govuk-form-group">
       <fieldset class="govuk-fieldset" aria-describedby="cookie-hint-1">
-        <legend class="govuk-fieldset__legend govuk-fieldset__legend--m">Cookies used to improve 'Check you can get legal aid'</legend>
+        <legend class="govuk-fieldset__legend govuk-fieldset__legend--m">{% trans %}Cookies used to improve 'Check you can get legal aid'{% endtrans %}</legend>
         <span id="cookie-hint-1" class="govuk-hint">
           <p>
-            We use Google Analytics to measure how you use ‘Check you can get legal aid’ so we can improve it based on user needs. 
-            We do not allow Google to use or share the data about how you use this site. 
+            {% trans %}We use Google Analytics to measure how you use ‘Check you can get legal aid’ so we can improve it based on user needs. 
+            We do not allow Google to use or share the data about how you use this site.{% endtrans %}
```
Comment (MalcolmVonMoJ) [inline cla_public/templates/_cookie-consent-form.html]: Should we split this up sentence by sentence to reduce the impact of whitespace and avoid whole paragraphs being a single translation?
Like this:
`{% trans %}We use Google Analytics to measure how you use ‘Check you can get legal aid’ so we can improve it based on user needs.{% endtrans %} `
`{% trans %}We do not allow Google to use or share the data about how you use this site.{% endtrans %} `

File: cla_public/templates/_cookie-popup.html [hunk 16]
```diff
@@ -1,18 +1,18 @@
...
+          <a class="gem-c-button gem-c-button--secondary gem-c-button--inline" role="button" data-module="track-click" data-track-category="cookieBanner" data-track-action="Cookie banner settings clicked" href="/cookie-settings">{% trans %}Cookie settings{% endtrans %}</a>
         </div>
       </div>
     </div>
 
   <div class="gem-c-cookie-banner__confirmation govuk-width-container" tabindex="-1">
     <p class="gem-c-cookie-banner__confirmation-message">
-      You’ve accepted all cookies. You can <a class="govuk-link" href="/cookie-settings" data-module="track-click" data-track-category="cookieBanner" data-track-action="Cookie banner settings clicked from confirmation">change your cookie settings</a> at any time.
+      {% trans %}You’ve accepted all cookies. You can {% endtrans %}<a class="govuk-link" href="/cookie-settings" data-module="track-click" data-track-category="cookieBanner" data-track-action="Cookie banner settings clicked from confirmation">{% trans %}change your cookie settings{% endtrans %}</a> {% trans %}at any time.{% endtrans %}
```
Comment (MalcolmVonMoJ) [inline cla_public/templates/_cookie-popup.html]: This should be one single translation.  To do it this way makes the assumption that Welsh word order is the same and I doubt that it is. 
`{% trans %}You’ve accepted all cookies. You can <a class="govuk-link" href="/cookie-settings" data-module="track-click" data-track-category="cookieBanner" data-track-action="Cookie banner settings clicked from confirmation">change your cookie settings</a> at any time.{% endtrans %}`

File: cla_public/templates/_cookie-consent-form.html (hunk 13 above)
Comment (MalcolmVonMoJ) [inline cla_public/templates/_cookie-consent-form.html]: Missing class - see above

File: cla_public/templates/_cookie-consent-form.html (hunk 14 above)
Comment (MalcolmVonMoJ) [inline cla_public/templates/_cookie-consent-form.html]: This class should not have been removed.

File: cla_public/static-templates/errors/50x.html (hunk 12 above)
Comment (exonian) [reply cla_public/static-templates/errors/50x.html]: For leading and trailing whitespace (ie this does not apply for the other comment, which addresses whitespace in between lines in the middle of a paragraph), I think that we can use `{%- trans -%}` to handle this. I might be wrong. Worth checking.

File: cla_public/templates/_cookie-consent-form.html (hunk 14 above)
Comment (joshlow) [reply cla_public/templates/_cookie-consent-form.html]: @MalcolmVonMoJ - can you suggest an alternative pls? These bullets are not currently showing on the live site: https://checklegalaid.service.gov.uk/cookie-settings

Comment (MalcolmVonMoJ) [reply cla_public/templates/_cookie-consent-form.html]: If bullets are wanted, use `<ul class="govuk-list govuk-list--bullet">`

Comment (said-moj) [reply cla_public/templates/_cookie-consent-form.html]: This has now been done

File: cla_public/static-templates/errors/50x.html [hunk 17]
```diff
@@ -109,7 +110,7 @@
           </div>
           <div class="govuk-header__content">
             <a href="/" class="govuk-header__link govuk-header__link--service-name">
-              Civil Legal Advice
+              {% trans %}Civil Legal Advice% endtrans %}
```
Comment (MalcolmVonMoJ) [inline cla_public/static-templates/errors/50x.html]: Missing `{`

**Pull request merged**